pandas/tests/strings/test_extract.py

```diff
@@ -51,163 +52,184 @@ def test_extract_expand_False():
-    exp = Series([np.nan, np.nan, np.nan], dtype=object)
-    tm.assert_series_equal(result, exp)
+    expected = Series([np.nan, np.nan, np.nan], dtype=any_string_dtype)
+    tm.assert_series_equal(result, expected)
 
     # two groups, no matches
     result = s.str.extract("(_)(_)", expand=False)
-    exp = DataFrame(
-        [[np.nan, np.nan], [np.nan, np.nan], [np.nan, np.nan]], dtype=object
+    expected = DataFrame(
+        [[np.nan, np.nan], [np.nan, np.nan], [np.nan, np.nan]], dtype=any_string_dtype
     )
-    tm.assert_frame_equal(result, exp)
+    tm.assert_frame_equal(result, expected)
 
     # one group, some matches
     result = s.str.extract("([AB])[123]", expand=False)
-    exp = Series(["A", "B", np.nan])
-    tm.assert_series_equal(result, exp)
+    expected = Series(["A", "B", np.nan], dtype=any_string_dtype)
+    tm.assert_series_equal(result, expected)
 
     # two groups, some matches
     result = s.str.extract("([AB])([123])", expand=False)
-    exp = DataFrame([["A", "1"], ["B", "2"], [np.nan, np.nan]])
-    tm.assert_frame_equal(result, exp)
+    expected = DataFrame(
+        [["A", "1"], ["B", "2"], [np.nan, np.nan]], dtype=any_string_dtype
+    )
+    tm.assert_frame_equal(result, expected)
 
     # one named group
     result = s.str.extract("(?P<letter>[AB])", expand=False)
-    exp = Series(["A", "B", np.nan], name="letter")
-    tm.assert_series_equal(result, exp)
+    expected = Series(["A", "B", np.nan], name="letter", dtype=any_string_dtype)
+    tm.assert_series_equal(result, expected)
 
     # two named groups
     result = s.str.extract("(?P<letter>[AB])(?P<number>[123])", expand=False)
-    exp = DataFrame(
-        [["A", "1"], ["B", "2"], [np.nan, np.nan]], columns=["letter", "number"]
+    expected = DataFrame(
+        [["A", "1"], ["B", "2"], [np.nan, np.nan]],
+        columns=["letter", "number"],
+        dtype=any_string_dtype,
     )
-    tm.assert_frame_equal(result, exp)
+    tm.assert_frame_equal(result, expected)
 
     # mix named and unnamed groups
     result = s.str.extract("([AB])(?P<number>[123])", expand=False)
-    exp = DataFrame([["A", "1"], ["B", "2"], [np.nan, np.nan]], columns=[0, "number"])
-    tm.assert_frame_equal(result, exp)
+    expected = DataFrame(
+        [["A", "1"], ["B", "2"], [np.nan, np.nan]],
+        columns=[0, "number"],
+        dtype=any_string_dtype,
+    )
+    tm.assert_frame_equal(result, expected)
 
     # one normal group, one non-capturing group
     result = s.str.extract("([AB])(?:[123])", expand=False)
-    exp = Series(["A", "B", np.nan])
-    tm.assert_series_equal(result, exp)
+    expected = Series(["A", "B", np.nan], dtype=any_string_dtype)
+    tm.assert_series_equal(result, expected)
 
     # two normal groups, one non-capturing group
-    result = Series(["A11", "B22", "C33"]).str.extract(
-        "([AB])([123])(?:[123])", expand=False
+    s = Series(["A11", "B22", "C33"], dtype=any_string_dtype)
```

Comment:
feel free to break up really long tests like this (followon), unless easy to view in the same one